src/Analytics/BasicProvider.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Message\Mothership\CMS\Analytics;
+
+/**
+ * @author Samuel Trangmar-Keates
+ *
+ * A basic provider class which is constructed with the reference, params and names
+ * required. Useful for basic providers.
+ *
+ * If there is a more complecated provider, then AnalyticsProviderInterface should
+ * be implemented
+ */
+class BasicProvider implements AnalyticsProvidorInterface
+{
+	/**
+	 * @var string
+	 */
+	private $_name;
+
+	/**
+	 * @var string
+	 */
+	private $_viewRef;
+
+	/**
+	 * @var array
+	 */
+	private $_params;
+
+	/**
+	 * Construct the object with the correct fields.
+	 * 
+	 * @param string $name    The object identifier
+	 * @param string $viewRef The view reference
+	 * @param array  $params  The parameters to pass to the view
+	 */
+	public function __construct($name, $viewRef, $params = [])
```

Comment:
The other way of doing it usually just straight up implements the interface. For instance in PageTypes the getViewReference() method typically looks like: 

```php
public function getViewReference()
{
  return 'Mothership:Site::page_type:home';
}
```

How do you mean you loose control of the constructor? If you needed to have control of the constructor (you're doing something weird) you can have it - write a new class extending this or implementing the interface.

The BasicProvider is for basic providers, where writing another class for something basic is overkill. From the point of someone actually using mothership, where they want to add a provider they can do that in one service->extends(). Whereas otherwise they would have to create an entirely new class -> effort.

I agree with the typehint and stacktrace comment, and think the way suggested is better. I will make a new class extending this, and suggest that this is the best practice way of doing it. The BasicProvider will be there still there so enables what is mentioned one paragraph up.